tools/cldr-code/src/test/java/org/unicode/cldr/test/TestCheckLogicalGroupings.java

```diff
@@ -79,4 +76,53 @@ private void assertCheckStatus(String localeID, CheckLogicalGroupings clg, Strin
                 () -> "Expected 0 errors but got " + possibleErrors2.get(0).toString() + " for " + xpath);
         }
     }
+
+    /**
+     * All non-optional paths that are in the same logical group should have the same coverage level
+     */
+    @Test
+    void testSameCoverageLevel() {
+        String[] locales = { "am", "en", "fr", "pt_PT", "zh" };
+        for (String localeId : locales) {
+            sameLevel(localeId);
+        }
+    }
+
+    void sameLevel(String localeId) {
+        final CLDRConfig config = CLDRConfig.getInstance();
+        final CLDRFile cldrFile = config.getCLDRFile(localeId, true);
+        final SupplementalDataInfo sdi = SupplementalDataInfo.getInstance();
+        final CoverageLevel2 coverageLevel = CoverageLevel2.getInstance(sdi, localeId);
+        for (final String path : cldrFile.fullIterable()) {
+            Set<String> grouping = LogicalGrouping.getPaths(cldrFile, path);
+            if (grouping != null && grouping.size() > 1) {
+                LogicalGrouping.removeOptionalPaths(grouping, cldrFile);
```

Comment:
This "removeOptionalPaths" is the only difference between this PR and the preceding draft PR, as far as the unit test is concerned

Reply:
also the test "if (grouping != null) … " was added.